source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -73,6 +74,25 @@ std::vector<Http::HeaderUtility::HeaderData> headerMatchers(const T& matcher_pro
   return matchers;
 }
 
+// Transforms the proto list of 'auth_scopes' into a vector of std::string, also
+// handling the default value logic
+template <class T> std::vector<std::string> authScopesList(const T& auth_scopes_protos) {
```

Comment:
I would just use the actual type (`Protobuf::RepeatedPtrField<std::string>`) here instead of making this generic

Reply:
@snowp Should I change to `Protobuf::RepeatedPtrField<std::string>` in the class property as well, right? Not only in the function return?

Reply:
@snowp is that what you mean? ` template <class T> Protobuf::RepeatedPtrField<std::string> authScopesList`

Reply:
I was thinking to remove the T from the paramter list and use the real type instead, so `std::vector<std::string> authScopesList(const Protobuf::RepeteadPtrField<std::string>& auth_scopes_proto)`. 

If we need to convert to vector<std::string>` then we'd keep that on the class, if not we don't even need this conversion function

Reply:
Got it. Added a commit to fix this.